syft/federated/plan.py

```diff
@@ -351,11 +352,20 @@ def _get_plan_output(self, result_ids, return_ptr=False):
         return responses
 
     def _execute_plan_locally(self, result_ids, *args, **kwargs):
+        args = self._keep_only_tensor_args(args)
         self._update_args(args, result_ids)
         self._execute_plan()
         responses = self._get_plan_output(result_ids)
         return responses
 
+    def _keep_only_tensor_args(self, args):
+        """Keeps only Tensors in args. This step is needed for method plans."""
+        tensor_args = []
+        for arg in args:
+            if isinstance(arg, torch.Tensor):
+                tensor_args.append(arg)
+        return tensor_args
+
```

Comment:
I'm not sure to understand why you need only to keep the tensor args, would you like to detail a bit more the dosctring?

Reply:
This is due to a weird behavior of plans that add self as first argument in the method case. Maybe we should check whether this code (plan.py, line 341)
```
        # Support for method hooked in plans
        if self.self is not None:
            args = [self.self] + list(args)
        return self.execute_plan(args, result_ids)
```
 is really necessary.

Reply:
I've removed this function and changed the behaviour to make more clear why do we need it

Reply:
Ok thanks I'll look at the changes. That's right we needed to include the .self as we call methods , but maybe there is a work around to make it more transparent - will check your latest changes